resources/views/oauth2/profile/edit-client.blade.php

```diff
@@ -1,171 +1,213 @@
+        <div class="row">
+            <div class="col-md-6 col-md-offset-8">
+                Welcome, <a target="_self"
+                            href="{!! URL::action("UserController@getProfile") !!}">{!!Auth::user()->getIdentifier()!!}</a>
+            </div>
+        </div>
+    @endif
+@stop
+
 @section('scripts')
-    {!! HTML::script("assets/moment/min/moment.min.js") !!}
-    <script type="application/javascript">
-
-        var dataClientUrls =
-        {
-            refresh: '{!!URL::action("Api\\ClientApiController@setRefreshTokenClient",array("id"=>$client->id, "use_refresh_token" => "@use_refresh_token"))!!}',
-            rotate: '{!!URL::action("Api\\ClientApiController@setRotateRefreshTokenPolicy",array("id"=>$client->id, 'rotate_refresh_token'=>'@rotate_refresh_token'))!!}',
-            update: '{!!URL::action("Api\\ClientApiController@update",array("id"=>$client->id))!!}',
-            add_public_key: '{!!URL::action("Api\\ClientPublicKeyApiController@_create",array("id"=>$client->id))!!}',
-            get_public_keys: '{!!URL::action("Api\\ClientPublicKeyApiController@getAll",array("id"=>$client->id))!!}',
-            delete_public_key: '{!!URL::action("Api\\ClientPublicKeyApiController@_delete",array("id" => $client->id, 'public_key_id'=> '@public_key_id'))!!}',
-            update_public_key: '{!!URL::action("Api\\ClientPublicKeyApiController@_update",array("id" => $client->id, 'public_key_id'=> '@public_key_id'))!!}',
-            fetchUsers: '{!!URL::action("Api\\UserApiController@getAll")!!}',
-        };
-
-        var oauth2_supported_algorithms =
-        {
-            sig_algorihtms:
-            {
-                mac: {!!Utils\ArrayUtils::toJson(OAuth2\OAuth2Protocol::$supported_signing_algorithms_hmac_sha2)!!},
-                rsa: {!!Utils\ArrayUtils::toJson(OAuth2\OAuth2Protocol::$supported_signing_algorithms_rsa)!!}
-            },
-            key_management_algorihtms: {!!Utils\ArrayUtils::toJson(OAuth2\OAuth2Protocol::$supported_key_management_algorithms)!!},
-            content_encryption_algorihtms:  {!!Utils\ArrayUtils::toJson(OAuth2\OAuth2Protocol::$supported_content_encryption_algorithms)!!}
-        };
-
-        var current_admin_users  = [];
-        var is_mine = {!! $client->isOwner(Auth::user())? 1:0 !!}
-
-        @foreach($client->getAdminUsers() as $user)
-        current_admin_users.push({
-            "id": {!!$user->getId()!!} ,
-            "value": "{!! $user->getFullName() !!}",
-            "first_name": "{!! $user->first_name !!}",
-            "last_name": "{!! $user->last_name !!}"
+    <script>
+        let adminUsers = [];
+        let scopes = [];
+        let selectedScopes = [];
+        let supportedSigningAlgorithms = [];
+        let supportedKeyManagementAlgorithms = [];
+        let supportedContentEncryptionAlgorithms = [];
+        let supportedTokenEndpointAuthMethods = [];
+        let supportedJSONWebKeyTypes = [];
+
+        const AppTypes = {
```

Comment:
add all logic on controller and pass json objects does not bloat the template with logic